src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -427,6 +427,10 @@ private void sendData(MPDbAdapter dbAdapter, MPDbAdapter.Table table, String[] u
                                 } catch (UnsupportedEncodingException e) {
                                     throw new RuntimeException("UTF not supported on this platform?", e);
                                 }
+                                if (mFailedRetries > 0) {
+                                    mFailedRetries = 0;
+                                    removeMessages(FLUSH_QUEUE);
```

Comment:
Yes, it does. Let's assume we remove this line. Now, imagine that a user calls `flush()` and there is already a FLUSH_QUEUE job pending because of a previous failed network request. If this time the network request goes through, we won't flush the queue again until that previously scheduled job is executed (could be in 30 minutes!). That means we will queue events for 30 minutes, and that's too much. We do not add new FLUSH_QUEUE events if there is already one pending. The only option is if the user manually calls `flush()`.